### ISSUE TITLE
1131 - Record Announcements - Create Form for Announcements

### ISSUE THREAD

**Issue by AntBush** [bcgov/CONN-CCBC-portal#1398]
This card is to create all the frontend/UI elements for editing Announcements, could be further split into primary and secondary but seems excessive at this point. 

## Add announcement
- [x] User can copy paste one URL into URL box - if more than one then don't allow them to save and provide feedback
- [x] User can choose any date in past or future for announcement date
- [x] User can choose other CCBC IDs that was included in this announcement
- [x] Drop down should not show the current application's CCBC ID
- [x] User can choose other CCBC IDs that was included in this announcement. Input field is a multiple value filter with chips, list of values is the CCBC Is (https://mui.com/material-ui/react-autocomplete/#multiple-values)
- [x] If using Material UI input, ensure input box is the same height & border as other input boxes
- [x] Form field top aligns as it expands
- [x] Is responsive to mobile/small width screens
- [x] Type is dropdown with options: `Primary` and `Secondary`

![image.png](https://images.zenhubusercontent.com/61ae5dab07758299775d9d78/9a89bcc4-75e2-4047-b1c9-119379886bfd)
![image.png](https://images.zenhubusercontent.com/61ae5dab07758299775d9d78/3d4e117c-b5cf-46f4-b24f-5989eb3c8f12)

- [x] Type, URL, & date are required. Outlined in red if empty on save.
- [x] Error message for incorrect URL appears below input field, "Invalid URL. Please copy & paste from your browser."
![image.png](https://images.zenhubusercontent.com/61ae5dab07758299775d9d78/49ad2719-3b9f-47b0-bc84-12b63165e836)

**Comment by MarsRomer** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/CCBC---Delivery-Board-61c0c1204c8bcb001491c5f3/issues/bcgov/CONN-CCBC-portal/1398?planning-poker) with Zenhub @AntBush @marcellmueller @rafasdc @VB-at-Bis

**Comment by AntBush** [comment]
https://codesandbox.io/s/wandering-platform-ys5j1k?file=/demo.tsx If it's any help with styling

**Comment by elliottSlee** [comment]
@marcellmueller Design changes:
- "Add announcement" removed
- Type field added
- To fit the Type field, `Announcement date` was changed to `Date announced` to narrow the field

**Comment by MarsRomer** [comment]
This looks great!  Just a couple of things that weren't explicit in the acceptance criteria that I am wondering about:
1. The app properly makes the text box border red for Announcement Type, URL , and date if it's not filled out or if the URL is not filled out correctly, however it doesn't turn it back to black border when it is subsequently filled out properly and stays red.  It's not a big deal, but I would expect the text box outline to turn black that as soon as I choose a value from the drop down in the Announcement type, and as soon as I pick the date, or correct the URL.
2. The other thing I noticed is that in the 'Other projects in announcement' box I love how if I type a couple of numbers it generates a dropdown list of matching IDs, however once I have one of the IDs added to that box, if I type more in the text box I don't see what I have typed.  Is there a way that we can see what was typed?  this might be more of a question for @elliottSlee.   This too is not  a big deal but would like to hear your thoughts

**Comment by elliottSlee** [comment]
@MarsRomer 
1. The pattern of validating whether it's correct after the correction is made is such a great UX and I'd love to have that. We need to think about how this auto-validation (not running on the click Save) will affect the experience of the first time they start filling it out. For example, if it's validating the second they start putting anything in, will it appear as an error? In the case of this Announcements form, it's not a big deal because type is a dropdown click, URL is copy & pasted, date is selected, and CCBC IDs are closed list so they are unlikely to be errors of data entry except maybe with the URL.
2. There's something weird about this that we need to fix. Not being able to see what I typed, nor being able to see the text cursor (the flashing line indicating where I am typing in the input box) caused me to press backspace more than the # of characters I had entered which removed a CCBC ID I had added. The sample MaterialUI version of this box shows the text being entered: https://mui.com/material-ui/react-autocomplete/#multiple-values
3. That should be solved with #1397
